### ISSUE TITLE
SP19 onwards Vue webpart does not start / build failed

### ISSUE THREAD

**Issue by SkipperRG** [pnp/generator-spfx#231]
#### General Information
* [x ] Usage
* [ ] Development
* [ ] Documentation
* [ ] Feature Request

#### Specific generator
* [ ] Addons
* [ ] HandlebarsJS

#### What Version you are running?
* NodeJS: 10.18.1
* NPM: 6.13.6
* @pnp/spfx: 1.11.0

#### Describe your problem

I generate a SP19 onwards webpart with Vue. All other generator-options i choose the default value.
After correcting the problem with the copy-assets.json file (see my other issue) i try to run the webpart using gulp serve.

Following error is displayed

> Error - typescript - src\webparts\helloWorld\HelloWorldWebPart.ts(14,32): error TS2307: Cannot find module './components/HelloWorld.vue'.

These file exists in the path, and the file shims-vue.d.ts exists although (some hints in the internet says that in this error condition these file may be missing).

Here the output from the installation and the generation of the webpart.

![InstallationGenerator](https://user-images.githubusercontent.com/60735014/73929119-921f1a80-48d4-11ea-9f62-63b2d763b091.jpg)

![Generatorlauf](https://user-images.githubusercontent.com/60735014/73929141-9814fb80-48d4-11ea-8f29-8912927eb4c1.jpg)

![Generatorlauf2](https://user-images.githubusercontent.com/60735014/73929906-f2628c00-48d5-11ea-8b79-4b31bc199373.jpg)

**Comment by StfBauer** [comment]
Thank you for reporting. Unfortunately there are currently some issue with the support for SharePoint 2019.

I did some digging and found a working solution. This will come to the next version of the generator.

You need to downgrade some of the dependencies:

```
{
  "name": "ansocht",
  "version": "0.0.1",
  "private": true,
  "engines": {
    "node": ">=0.10.0"
  },
  "scripts": {
    "build": "gulp bundle",
    "clean": "gulp clean",
    "test": "gulp test"
  },
  "dependencies": {
    "@microsoft/sp-core-library": "1.6.0",
    "@microsoft/sp-lodash-subset": "1.6.0",
    "@microsoft/sp-office-ui-fabric-core": "1.6.0",
    "@microsoft/sp-webpart-base": "1.6.0",
    "@types/es6-promise": "0.0.33",
    "@types/webpack-env": "1.13.1",
    "vue": "^2.6.11",
    "vue-class-component": "^7.2.2",
    "vue-property-decorator": "^8.4.0"
  },
  "devDependencies": {
    "@microsoft/sp-build-web": "1.6.0",
    "@microsoft/sp-module-interfaces": "1.6.0",
    "@microsoft/sp-webpart-workbench": "1.6.0",
    "@types/chai": "3.4.34",
    "@types/mocha": "2.2.38",
    "ajv": "~5.2.2",
    "fork-ts-checker-webpack-plugin": "^0.4.9",
    "gulp": "~3.9.1",
    "sass-loader": "^7.1.0",
    "ts-loader": "^6.2.1",
    "tslint-microsoft-contrib": "~5.0.0",
    "vue-loader": "^15.2.4",
    "vue-template-compiler": "^2.5.16"
  }
}
```

This were the dependencies which succeeded for me.

In addition your copy-assets.json only needs to look like this:

```json
{
  "$schema": "https://developer.microsoft.com/json-schemas/spfx-build/copy-assets.schema.json",
  "deployCdnPath": "temp/deploy"
}
```

Just to make sure it works the same way I copied the gulp file that worked for me:
```javascript
'use strict';

const gulp = require('gulp');
const build = require('@microsoft/sp-build-web');

build.addSuppression(`Warning - [sass] The local CSS class 'ms-Grid' is not camelCase and will not be type-safe.`);

// Merge custom loader to web pack configuration

const VueLoaderPlugin = require('vue-loader/lib/plugin');
const vuePlugin = new VueLoaderPlugin();
const themedStyleLoader = require.resolve('@microsoft/loader-load-themed-styles');



    build.configureWebpack.mergeConfig({

        additionalConfiguration: (generatedConfiguration) => {

            const loadersConfigs = [{
                test: /\.vue$/, // vue
                use: [{
                    loader: 'vue-loader'
                }]
            }, {
                resourceQuery: /vue&type=script&lang=ts/, // typescript
                loader: 'ts-loader',
                options: {
                    appendTsSuffixTo: [/\.vue$/],
                    transpileOnly: true
                }
            }, {
                resourceQuery: /vue&type=style.*&lang=scss/, // scss
                use: [{
                    loader: themedStyleLoader,
                    options: {
                        async: true
                    }
                },
                {
                    loader: 'css-loader',
                    options: {
                        modules: true,
                        localIdentName: '[local]_[sha1:hash:hex:8]'
                    }
                },
                    'sass-loader'
                ]
            }];

            generatedConfiguration.plugins.push(vuePlugin);
        
                generatedConfiguration.module.rules.push(...loadersConfigs);

            return generatedConfiguration;

        }

    });

// register custom watch for Vue.JS files
// copy of '.vue' files will be handled by 'copy-static-assets.json'
gulp.watch('./src/**/*.vue', event => {
    // copy empty index.ts onto itself to launch build procees
    gulp.src('./src/index.ts')
        .pipe(gulp.dest('./src/'));

});


build.initialize(gulp);
```

This goes way back to @pnp/spfx version 1.4.3.

**Comment by SkipperRG** [comment]
Thank you very much, now the error is gone. 
But now i have another problem, i am trying to solve it, but next week i am not in the office.
I will contact you again when I know more.
Have a nice weekend!

**Comment by StfBauer** [comment]
@SkipperRG Related to this or something else. I would prefer if you could open another issue when it is not directly related.

**Comment by StfBauer** [comment]
Hi @SkipperRG I checked the older versions of the generator and the issue was always persistent. Sadly the code above is coming from a SPO project but not form an On-Premises version.

On-premises 2019 uses SPFx 1.4.1 - there are some issues in the whole TypeScript handling which was solved only in 1.6.0 of the framework.

I am really sad but I removed VueJS support form on-prem 2019 project for now.
